packages/core/state/interaction/logics.ts

```diff
@@ -499,6 +501,105 @@ const openWithLogic = createLogic({
     type: OPEN_WITH,
 });
 
+/**
+ * Interceptor responsible for translating an EDIT_FILES action into a progress tracked
+ * series of edits on the files currently selected.
+ */
+const editFilesLogic = createLogic({
+    async process(deps: ReduxLogicDeps, dispatch, done) {
+        const fileService = interactionSelectors.getFileService(deps.getState());
+        const fileSelection = selection.selectors.getFileSelection(deps.getState());
+        const sortColumn = selection.selectors.getSortColumn(deps.getState());
+        const annotationNameToAnnotationMap = metadata.selectors.getAnnotationNameToAnnotationMap(
+            deps.getState()
+        );
+        const {
+            payload: { annotations, filters },
+        } = deps.action as EditFiles;
+
+        // Gather up the files for the files selected currently
+        // if filters is present then actual "selected" files
+        // are the ones that match the filters, this happens when
+        // editing a whole folder for example
+        let filesSelected;
+        if (filters) {
+            const fileSet = new FileSet({
+                filters,
+                fileService,
+                sort: sortColumn,
+            });
+            const totalFileCount = await fileSet.fetchTotalCount();
+            filesSelected = await fileSet.fetchFileRange(0, totalFileCount);
+        } else {
+            filesSelected = await fileSelection.fetchAllDetails();
+        }
+
+        // Break files into batches of 10 File IDs
+        const fileIds = filesSelected.map((file) => file.uid);
+        const batches = chunk(fileIds, 10);
+
+        // Dispatch an event to alert the user of the start of the process
+        const editRequestId = uniqueId();
+        const editProcessMsg = "Editing files in progress.";
+        // TODO: Lyndsay's design did not include a progress bar for editing files
+        // nor a way to cancel the process. This is a placeholder for now.
+        dispatch(processStart(editRequestId, editProcessMsg, noop));
```

Comment:
hmm... ig thats fine perhaps drop the TODO then?

Reply:
removed in [e885706](https://github.com/AllenInstitute/biofile-finder/pull/372/commits/e885706c8a2a107d4fd473cacbeb720e0c5c5088)